lib/core/wireprotocol/command.js

```diff
@@ -118,4 +131,40 @@ function supportsOpMsg(topologyOrServer) {
   return description.maxWireVersion >= 6 && description.__nodejs_mock_server__ == null;
 }
 
+function _cryptCommand(server, ns, cmd, options, callback) {
+  const shouldBypassAutoEncryption = !!server.s.options.bypassAutoEncryption;
+  const autoEncrypter = server.s.options.autoEncrypter;
+  function commandResponseHandler(err, response) {
+    if (err || response == null) {
+      callback(err, response);
+      return;
+    }
+
+    autoEncrypter.decrypt(response.result, (err, decrypted) => {
+      if (err) {
+        callback(err, null);
+        return;
+      }
+
+      response.result = decrypted;
+      response.message.documents = [decrypted];
+      callback(null, response);
+    });
+  }
+
+  if (shouldBypassAutoEncryption) {
+    _command(server, ns, cmd, options, commandResponseHandler);
```

Comment:
q: is the implication of this that we will send an unencrypted message, but will receive back an encrypted response?

Reply:
Yes. Imagine you only want to `find` a series of encrypted documents, you don't want to pay the price of encrypting the command because there's nothing to encrypt.